frb_codegen/src/utils/misc.rs

```diff
@@ -184,3 +185,18 @@ macro_rules! ir {
         $crate::ir!($($rest)*);
     }
 }
+
+pub fn read_rust_file(path: &PathBuf) -> String {
+    let module = match path.file_name().and_then(|e| e.to_str()).unwrap_or("") {
+        "mod.rs" => path
+            .parent()
+            .and_then(|p| p.file_name())
+            .and_then(|f| f.to_str()),
+        "lib.rs" => None,
+        "" => None,
+        _ => path.file_stem().and_then(|f| f.to_str()),
```

Comment:
nit: what if api.rs is in a nested module? e.g. `src/hello/world/api.rs`

Reply:
good point. I should split the path before the  "src" and create a module like hello::world::api in that case.
This also means adjustments in the extract_module function

Reply:
sub modules are correctly found now. Also added some tests